### PR TITLE
bugfix/16686-vbp-yaxis-extremes 

### LINKED ISSUE #16686: VBP with compare - strange yAxis extremes when moving the navigator
Like in the title.

vbp-bug

#### Live demo with steps to reproduce
Take an official VBP indicator demo, set `compare: 'percent'` on the VBP series and then move the navigator
https://jsfiddle.net/zcy6vp8e/

#### Product version
Highcharts Stock v9.3.1

### PR DESCRIPTION
**Pull request by kamil-musialowski**
Fixed #16686, yAxis extremes were incorrect with VBP indicator compare.

Although it looks like a hacky solution, VBP (with compare) recalculations should not affect the yAxis extremes calculations, since due to the VBP position and rotation, `yData` does not affect the `yAxis` ticks.

### CODE REVIEW

**Review by pawelfus — CHANGES_REQUESTED** [review]
A bit too hacky :) Let's keep VBP Indicator code inside its module. `compare` (and `cumulative`) options should be ignored by VBP. Perhaps it's possible to remove/disable these options in `init()`?

**Comment by kamil-musialowski** [comment]
After #16277 I think it should be possible to use it in VBP, so I don't think we should get rid of the compare option, what do you think?

**Comment by pawelfus** [comment]
Good point! We should keep it. Now I hope for an implementation similar to this workaround: https://jsfiddle.net/BlackLabel/c2dohrta/ - but again, it looks a little hacky.

The best would be to simply.. remove the attached event for this particular series. Something like this:
```ts
// VBPIndicator.ts
removeEvent(VbPSeries, 'afterGetExtremes', DataModifyComposition.afterGetExtremes);
```

But I'm not sure if something similar is supported or should be avoided (because it would load the whole DataModify composition into the VbP indicator). Perhaps @bre1470 or @TorsteinHonsi can bring the answer to light

**Comment by pawelfus** [comment]
Let's try it and see how many bytes will be added to `indicators-all.js`. Could you try it @kamil-musialowski?

**Comment by kamil-musialowski** [comment]
Sure thing, I'll try it :)

**Comment by kamil-musialowski** [comment]
After discussing the case with @pawelfus, we have discovered that it won't be possible to remove the event (since it's added on the Series prototype, not on the VBP prototype). Therefore, we have agreed to try the solution with overwriting the `getExtremes` method in VBP class.

File: ts/Stock/Indicators/VBP/VBPIndicator.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ const {
 } = SeriesRegistry.seriesTypes;
 import U from '../../../Core/Utilities.js';
 import StockChart from '../../../Core/Chart/StockChart.js';
+import DataExtremesObject from '../../../Core/Series/DataExtremesObject';
```
Comment (pawelfus) [inline ts/Stock/Indicators/VBP/VBPIndicator.ts:49]: Import type only

**Review by pawelfus — CHANGES_REQUESTED** [review]
Tiny nitpicking, we can go with this solution, thanks!

File: ts/Stock/Indicators/VBP/VBPIndicator.ts (hunk 1 above)
Comment (kamil-musialowski) [reply ts/Stock/Indicators/VBP/VBPIndicator.ts:49]: Oh, I must have missed this one - thanks!

**Review by pawelfus — APPROVED** [review]
Thanks!

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**